File: app/src/main/java/com/learnteachcenter/ltcreikiclock/ui/SplashActivity.java
package com.learnteachcenter.ltcreikiclock.ui;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.auth.GetTokenResult;
import com.learnteachcenter.ltcreikiclock.R;
import com.learnteachcenter.ltcreikiclock.auth.LoginActivity;
import com.learnteachcenter.ltcreikiclock.auth.SessionManager;
import com.learnteachcenter.ltcreikiclock.ui.reiki.ReikiListActivity;

public class SplashActivity extends BaseActivity {

    private static final String TAG = "Reiki";

    private SessionManager mSession;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        // Check session
        mSession = new SessionManager(getApplicationContext());
        mSession.checkLoginAndRedirect();
    }
}

File: app/src/main/java/com/learnteachcenter/ltcreikiclock/ui/splash/SplashActivity.kt
package com.learnteachcenter.ltcreikiclock.ui.splash

import android.content.Intent
import android.os.Bundle

import com.learnteachcenter.ltcreikiclock.R
import com.learnteachcenter.ltcreikiclock.application.ReikiApplication
import com.learnteachcenter.ltcreikiclock.ui.login.LoginActivity
import com.learnteachcenter.ltcreikiclock.authentication.FirebaseAuthenticationInterface
import com.learnteachcenter.ltcreikiclock.ui.base.BaseActivity
import com.learnteachcenter.ltcreikiclock.ui.reiki.ReikiListActivity
import javax.inject.Inject

class SplashActivity : BaseActivity() {

    @Inject
    lateinit var authentication: FirebaseAuthenticationInterface;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        (application as ReikiApplication).appComponent.inject(this)

        val i: Intent;

        if (authentication.isSignedIn()) {
            i = Intent(applicationContext, ReikiListActivity::class.java)
        } else {
            i = Intent(applicationContext, LoginActivity::class.java)
        }

        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
        i.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        startActivity(i)
    }

    companion object {
        private val TAG = "Reiki"
    }
}
